script.js

```diff
@@ -1,4 +1,709 @@
+  }
+};
+
+// Find the running sum of the numbers in an array
+var addTotal = function (array) {
+  var total = ``;
+  counter = 0;
+  while (counter < array.length) {
+    total = Number(total) + Number(array[counter]);
+    counter += 1;
+  }
+  return total;
+};
+
+// Find winning player based on result mode (normal/reversed)
+// Set 'Player 1' as default winner
+var winnersGreatest = [1];
+var winnersSmallest = [1];
+var findWinner = function (array, normalOrReversed) {
+  // Generate updated list of winning players
+  if (
+    // Do not need to compare if the player is already in the winners list since the previous round
+    !winnersGreatest.includes(playerNo + 1) &&
+    // Add to list if the player's number is same as that of current winner(s)
+    array[[winnersGreatest[0]] - 1] == array[playerNo]
+  ) {
+    console.log("A", array[[winnersGreatest[0]] - 1]);
+    console.log("B", array[playerNo]);
+    console.log("C", winnersGreatest);
+    winnersGreatest.push(playerNo + 1);
+    console.log("D", winnersGreatest);
+    // Replace winner list if there is a new winner
+  } else if (array[[winnersGreatest[0]] - 1] < array[playerNo]) {
+    winnersGreatest = [];
+    winnersGreatest.push(playerNo + 1);
+    console.log("D", winnersGreatest);
+  }
+  if (
+    // Do not need to compare if the player is already in the winners list since the previous round
+    !winnersSmallest.includes(playerNo + 1) &&
+    // Add to list if the player's number is same as that of current winner(s)
+    array[[winnersSmallest[0]] - 1] == array[playerNo]
+  ) {
+    winnersSmallest.push(playerNo + 1);
+    // Replace winner list if there is a new winner
+  } else if (array[[winnersSmallest[0]] - 1] > array[playerNo]) {
+    winnersSmallest = [];
+    winnersSmallest.push(playerNo + 1);
+  }
+  // Return winners with greatest/smallest total depending on game mode (normal/reversed)
+  if (normalOrReversed == NORMAL) {
+    return winnersGreatest;
+  }
+  if (normalOrReversed == REVERSED) {
+    return winnersSmallest;
+  }
+};
+
+// ***** MAIN FUNCTION *****
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log("player number", playerNo);
+  // Step 1: Check for instructions to change result mode (normal/reversed) first
+  if (input.toLowerCase() == REVERSED) {
+    resultMode = REVERSED;
+    // Reset Arrays to start new game in changed mode
+    competingNumbers = [];
+    competingArrays = [];
+    competingTotals = [];
+    return "You are now in reversed mode where the smaller number wins! <br> <br> Please enter the number of players for this game.";
+  }
+  if (input.toLowerCase() == NORMAL) {
+    resultMode = NORMAL;
+    // Reset Arrays to start new game in changed mode
+    competingNumbers = [];
+    competingArrays = [];
+    competingTotals = [];
+    return "You are now in normal mode where the bigger number wins! <br> <br> Please enter the number of players for this game.";
```

Comment:
It would be good to have a restart function of sorts! So as to repeat code lesser times!